plugins/org.eclipse.glsp.server/src/org/eclipse/glsp/server/internal/util/ReflectionUtil.java

```diff
@@ -13,7 +13,7 @@
  *
  * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
  ********************************************************************************/
-package org.eclipse.glsp.api.utils;
+package org.eclipse.glsp.server.internal.util;
```

Comment:
Since we encourage reflection in the GLSPModule (By only declaring the Handler, and making the handled Operation implicit), I think client applications might need this helper as well. I'm not 100% sure though; as the use case we had in Ecore was a temporary workaround, that I can probably revert now (We had to provide a custom version of the DIOperationHandlerRegistry)

Reply:
I'd say that this is util that typical adopters shouldn't have to use. I agree that it might be useful in certain cases, but at the same time I'd like to minimize the public API to avoid overhead when changing the API in the future.
What do you think, @martin-fleck-at and @tortmayr?

Reply:
I agree, this s a util that typical adopters shouldn't  have to use. However, I don't have a strong opinion wether we should make it public or not. I can see that it might be useful in some edge cases. And I don't think that the `ReflectionUtil` will see much changes in the future so the overhead should be minimal as well. I'm fine with both keeping it as is or exposing it.

Reply:
I feel the same as Tobias although I am slightly tilting towards making it public as the API will be very stable and it might be useful for some adopters to use it instead of writing the same code again. Of course I am aware that this is not "core" GLSP and making this "our" API is a bit unexpected. So I am fine with anything in this regard, I am sure that helped a lot :P

Reply:
In this case, the middle-ground is typically to export the package, but still mark it internal / hidden, so clients can use it with a nice warning. In this case, we don't have to guarantee the API, and clients that really need it have to think carefully about what they're doing; but still they don't have to rewrite it if they use it. And if it changes in the future, these clients would likely break anyway (Whether they use the exact same class, or a copy of it), as it probably means we changed the way e.g. Handlers are injected.

Reply:
Yes, I'd actually do that for all non-API classes; that is exporting them, but marking them as internal. There is no reason to really prevent an adopter to use those classes, but it is merely to communicate that this class may change anytime.
Thanks all for your feedback!